ex6/r_kobayashi/main.py

```diff
@@ -0,0 +1,196 @@
+import argparse
+import os
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
+
+
+def open_csv(fname):
+    """
+    open csv file
+
+    Parameter
+    ---------
+    fname : str
+        file name
+        
+    Return
+    ------
+    data : numpy.array
+        opened file in the form of numpy
+    """
+    data = pd.read_csv(fname, header=None)
+    data = np.array(data)
+
+    return data
+
+
+def standardize_data(data):
+    """
+    standardize data
+
+    Parameter
+    ---------
+    data : numpy.ndarray
+        observed data
+
+    Return
+    ------
+    s_data : numpy.ndarray
+        standardized data
+    """
+    m_data = np.mean(data, axis=0)
+    var_data = np.std(data, axis=0, ddof=1)
+    s_data = (data-m_data) / var_data
+
+    return s_data
+
+
+def pca(data):
+    """
+    calculate pca
+
+    Parameter
+    ---------
+    data : numpy.ndarray
+        standardized data
+
+    Returns
+    -------
+    sort_ev : numpy.ndarray
+        sorted eigenvector
+    contribution : np.ndarray
+        contribution rate
+    """
+    # 分散共分散行列を計算
+    cov = np.cov(data, rowvar=False)
+    # 固有値、固有ベクトルを計算
+    lam, ev = np.linalg.eigh(cov)
+    # 固有値、固有ベクトルを降順にソート
+    sort_lam = np.sort(lam)[::-1]
+    sort_ev = ev[:, np.argsort(lam)[::-1]]
+    contribution = sort_lam / np.sum(sort_lam)
+
+    return sort_ev, contribution
+
+
+def plot_2d(data, ev, contribution, graph_title):
+    """
+    plot 2d data
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        observed data
+    ev : numpy.ndarray
+        eigenvector
+    contribution : np.ndarray
+        contribution rate
+    graph_title : str
+        graph title
+    """
+    x = data[:, 0]
+    y = data[:, 1]
+    x_label = np.linspace(min(data[:, 0]), max(data[:, 0]), 100)
+    z1 = (ev[1, 0] / ev[0, 0]) * x_label
+    z2 = (ev[1, 1] / ev[0, 1]) * x_label
+
+    plt.figure(figsize=(10,8))
+    plt.scatter(x, y, c="m", label="data")
+    plt.plot(x_label, z1, c="b", label="Contribution rate: "+str(round(contribution[0], 3)))

```

Comment:
fstringを使うといいよ
参考：https://note.nkmk.me/python-f-strings/
```suggestion
    plt.plot(x_label, z1, c="b", label=f"Contribution rate: {round(contribution[0], 3)}")
```